packages/component-meta/scripts/generateComponentMeta.ts

```diff
@@ -221,9 +221,31 @@ const generateComponentMeta = (): void => {
     }
 
     let [, rawPropTypes] = /const [a-z][a-zA-z]+: (?:Omit<)?PropTypes<.+?> = ({[\s\S]+?});/.exec(source) || [];
+
+    // handle shared props
+    let sharedPropsExportFile; // also needed for import paths of variables
+    if (rawPropTypes && rawPropTypes.match(/\.{3}[a-zA-Z]+,/g)) {
+      const [, sharedPropsConstName] = rawPropTypes.match(/(shared[a-zA-z]+Types)/);
+      const [, sharedPropsImportPath] = source.match(new RegExp(`${sharedPropsConstName}[\\s\\S]+?from '(.+)';`)) || [];
+      const componentFilePath = componentFiles.find((file) =>
```

Comment:
This is kind of duplicated with the other location right? Why not make it reuasble?